src/test/java/dog/pawbook/logic/parser/EditDogCommandParserTest.java

```diff
@@ -0,0 +1,211 @@
+import dog.pawbook.model.managedentity.dog.Sex;
+import dog.pawbook.model.managedentity.tag.Tag;
+import dog.pawbook.testutil.EditDogDescriptorBuilder;
+
+public class EditDogCommandParserTest {
+
+    private static final String MESSAGE_INVALID_FORMAT =
+            String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditDogCommand.MESSAGE_USAGE);
+
+    private EditDogCommandParser parser = new EditDogCommandParser();
+
+    @Test
+    public void parse_missingParts_failure() {
+        // no index specified
+        assertParseFailure(parser, VALID_NAME_ASHER, MESSAGE_INVALID_FORMAT);
+
+        // no field specified
+        assertParseFailure(parser, "1", EditDogCommand.MESSAGE_NOT_EDITED);
+
+        // no index and no field specified
+        assertParseFailure(parser, "", MESSAGE_INVALID_FORMAT);
+    }
+
+    @Test
+    public void parse_invalidPreamble_failure() {
+        // negative index
+        assertParseFailure(parser, "-5" + NAME_DESC_ASHER, MESSAGE_INVALID_FORMAT);
+
+        // zero index
+        assertParseFailure(parser, "0" + NAME_DESC_ASHER, MESSAGE_INVALID_FORMAT);
+
+        // invalid arguments being parsed as preamble
+        assertParseFailure(parser, "1 some random string", MESSAGE_INVALID_FORMAT);
+
+        // invalid prefix being parsed as preamble
+        assertParseFailure(parser, "1 i/ string", MESSAGE_INVALID_FORMAT);
+    }
+
+    @Test
+    public void parse_invalidValue_failure() {
+        assertParseFailure(parser, "1" + INVALID_NAME_DESC, Name.MESSAGE_CONSTRAINTS); // invalid name
+        assertParseFailure(parser, "1" + INVALID_BREED_DESC, Breed.MESSAGE_CONSTRAINTS); // invalid breed
+        assertParseFailure(parser, "1" + INVALID_SEX_DESC, Sex.MESSAGE_CONSTRAINTS); // invalid sex
+        assertParseFailure(parser, "1" + INVALID_TAG_DESC, Tag.MESSAGE_CONSTRAINTS); // invalid tag
+        // invalid dateOfBirth
+        assertParseFailure(parser, "1" + INVALID_DATEOFBIRTH_DESC, DateOfBirth.MESSAGE_CONSTRAINTS);
+
+        // invalid breed followed by valid dob
+        assertParseFailure(parser, "1" + INVALID_BREED_DESC + DATEOFBIRTH_DESC_ASHER, Breed.MESSAGE_CONSTRAINTS);
+
+        // valid breed followed by invalid breed. The test case for invalid breed followed by valid breed
+        // is tested at {@code parse_invalidValueFollowedByValidValue_success()}
+        assertParseFailure(parser, "1" + BREED_DESC_BELL + INVALID_BREED_DESC, Breed.MESSAGE_CONSTRAINTS);
+
+        // while parsing {@code PREFIX_TAG} alone will reset the tags of the {@code Owner} being edited,
+        // parsing it together with a valid tag results in error
+        assertParseFailure(parser, "1" + TAG_DESC_FRIENDLY + TAG_DESC_QUIET + TAG_EMPTY, Tag.MESSAGE_CONSTRAINTS);
+        assertParseFailure(parser, "1" + TAG_DESC_FRIENDLY + TAG_EMPTY + TAG_DESC_QUIET, Tag.MESSAGE_CONSTRAINTS);
+        assertParseFailure(parser, "1" + TAG_EMPTY + TAG_DESC_FRIENDLY + TAG_DESC_QUIET, Tag.MESSAGE_CONSTRAINTS);
+
```

Comment:
how about invalid values for the other attribute fields like sex?

Reply:
Other fields are at the top right?

Reply:
my bad, do standardise how the comments are placed for the block of code, if i'm making this mistake it means the assessors might also make similar mistakes, where you won't be able to explain yourself anymore

Reply:
Done!